k9mail-library/src/main/java/com/fsck/k9/mail/Folder.java

```diff
@@ -113,7 +113,7 @@ public abstract void setFlags(List<? extends Message> messages, Set<Flag> flags,
 
     public abstract String getUidFromMessageId(Message message) throws MessagingException;
 
-    public void expunge() throws MessagingException
+    public void expunge(List<Long> knownDeletedUids) throws MessagingException
```

Comment:
With the `Folder` abstraction UIDs are opaque strings. UIDs being numbers is an implementation detail of `ImapFolder`. So until we can replace this interface with something more flexible let's keep this interface clean.

Also, sometimes (e.g. when emptying the Trash folder) an unconstrained EXPUNGE makes sense. So I'd keep the original method and add a new one that better describes what it is about. My suggestion:

```java
public void expungeUids(List<String> uids) throws MessagingException
```